torch/utils/checkpoint.py

```diff
@@ -39,59 +39,68 @@ def backward(ctx, *args):
         return (None,) + tuple(inp.grad for inp in detached_inputs)
 
 
-def checkpoint(run_function, *args):
+def checkpoint(function, *args):
     r"""Checkpoint a model or part of the model
 
-    Checkpoint works by trading compute for memory. It can be applied on any
-    part of the model. In the forward pass, the model activations are not
-    stored. The forward pass save the inputs tuple and the run_function
-    parameter. In the backwards pass, the saved inputs and run_function is
-    retreived, and the forward pass is done on the model again (non-volatile
-    this time) since we need to get the activations values for calculating the
-    gradient and then the gradients are calculated.
+    Checkpointing works by trading compute for memory. Rather than storing all
+    intermediate activations of the entire computation graph for computing
+    backward, the checkpointed part does **not** save intermediate activations,
+    and instead computes them in backward pass of this particular part. It can
+    be applied on any part of a model.
```

Comment:
nit: you used the word "part" 3 times in two sentences.

Also, I'd say it "recomputes" them to emphasise that it's extra compute.